crates/turborepo/tests/common/mod.rs

```diff
@@ -0,0 +1,60 @@
+use std::{path::Path, process::Command};
+
+use camino::Utf8Path;
+
+pub fn setup_fixture(
+    fixture: &str,
+    package_manager: &str,
+    test_dir: &Path,
+) -> Result<(), anyhow::Error> {
+    let script_path = Utf8Path::new(env!("CARGO_MANIFEST_DIR"))
+        .join("../../turborepo-tests/helpers/setup_integration_test.sh");
+
+    Command::new("bash")
+        .arg("-c")
+        .arg(format!("{} {} {}", script_path, fixture, package_manager))
+        .current_dir(test_dir)
+        .spawn()?
+        .wait()?;
+
+    Ok(())
+}
+
+/// Executes a command with different arguments in a specific fixture and
+/// package manager. Creates a snapshot file for each set of arguments
+#[macro_export]
+macro_rules! check {
+    ($fixture:expr, $package_manager:expr, $command:expr, $($name:expr => $query:expr,)*) => {
+        {
+            let tempdir = tempfile::tempdir()?;
+            crate::common::setup_fixture($fixture, $package_manager, tempdir.path())?;
+            $(
+                let output = assert_cmd::Command::cargo_bin("turbo")?
+                    .arg($command)
+                    .arg($query)
+                    .current_dir(tempdir.path())
+                    .output()?;
+
+                let stdout = String::from_utf8(output.stdout)?;
+                let query_output: serde_json::Value = serde_json::from_str(&stdout)?;
+                let test_name = format!(
+                    "{}_{}_({})",
+                    $fixture,
+                    $name.replace(' ', "_"),
+                    $package_manager
+                );
+
+                if !output.stderr.is_empty() {
+                    let stderr = String::from_utf8_lossy(&output.stderr);
+                    let stderr_test_name = format!(
+                        "{}_err",
+                        test_name
+                    );
+                    insta::assert_snapshot!(stderr_test_name, stderr);
```

Comment:
I don't think we want to snapshot stderr since it prints the turbo version. Can you write a custom assertion that makes sure we print version etc?

Reply:
yeah I'm gonna cut stderr snapshotting for now. When we need it I'll add it